vulnerabilities/data_source.py

```diff
@@ -57,16 +68,14 @@ class Advisory:
     summary: str
     impacted_package_urls: Iterable[PackageURL]
     resolved_package_urls: Iterable[PackageURL] = dataclasses.field(default_factory=list)
-    reference_urls: Sequence[str] = dataclasses.field(default_factory=list)
-    reference_ids: Sequence[str] = dataclasses.field(default_factory=list)
+    vuln_references: List[VulnerabilityReferenceUnit] = dataclasses.field(default_factory=list)
     cve_id: Optional[str] = None
 
     def __hash__(self):
```

Comment:
@pombredanne  and @haikoschol  what's the best way to handle this ? The this is being done atm is obviously not the idel way.

Reply:
VulnerabilityReferenceUnit is not my preferred name, but your design above looks OK.

Reply:
> Why do you think you need a hash here? 

Not 100% sure as @haikoschol  implemented this, probably because we want to put multiple Advisory's in  a set to avoid duplicates. 

I think Advisory can be made immutable with no pain, that way we won't need to expicitly implement __hash__ .

Reply:
The hash is there for a reason :) . 

I'm getting this  even after making Advisory immutable(by passing frozen=True in the dataclass decorator.) 

```
  File "/home/shivam/coding/opensource/vulnerablecode/vulnerabilities/data_source.py", line 191, in batch_advisories
    yield set(b)
  File "<string>", line 3, in __hash__
TypeError: unhashable type: 'set'

```

Reply:
@sbs2001 avoid using set. They are unordered and are seldom needed and you will no longer need hashing override

Reply:
@pombredanne  order doesn't matter in many places, since all Advisories will be dumped to database, it is just an intermediate data structure. Having said that, we really don't to have duplicate Advisories as those will cause IntegrityError